src/Mod/SoldierNamePool.cpp

```diff
@@ -129,6 +143,36 @@ std::string SoldierNamePool::genName(SoldierGender *gender, int femaleFrequency)
 	return name.str();
 }
 
+/**
+ * Returns a new random callsign from the
+ * lists of names contained within.
+ * @param gender Returned gender of the callsign.
+ * @return The soldier's callsign.
+ */
+std::string SoldierNamePool::genCallsign(const SoldierGender gender) const
+{
+	std::ostringstream callsign;
+	if (!_femaleCallsign.empty())
```

Comment:
When it will be empty?

Reply:
When user's only defined the male names. Same fallback mechanism as with last names.